src/configuration.ts

```diff
@@ -224,14 +224,19 @@ export function loadConfigurationFromPath(configFilePath?: string, originalFileP
             delete (require.cache as { [key: string]: any })[resolvedConfigFilePath];
         }
 
+        // assign defaultSeverity of parent if child configuration has no defaultSeverity set:
+        const parentDefaultSeverity = parentConfig !== undefined ? parentConfig.defaultSeverity : undefined;
+        rawConfigFile.defaultSeverity =
+            rawConfigFile.defaultSeverity !== undefined ? rawConfigFile.defaultSeverity : parentDefaultSeverity;
```

Comment:
IMO a `defaultSeverity` specified in the extending config file should always override the `defaultSeverity` in all extended configs.

Reply:
Yep, that's should still be the case with my additions.

Reply:
Let's consider the following:
* a.json has no defaultSeverity
* b.json has defaultSeverity "warning"
* c.json extends a.json and b.json and has defaultSeverity "error"

The result is:
* rules specified in c.json get a default of "error"
* rules from b.json get a default of "warning", because b.json specifies its own defaultSeverity
* rules from a.json get a default of "error" from c.json, because a.json has no defaultSeverity.

To override the defaultSeverity in b.json, you'd need to adjust this line to
```ts
rawConfigFile.defaultSeverity = parentDefaultSeverity !== undefined ? parentDefaultSeverity : rawConfigFile.defaultSeverity;
```

To make it clearer what's happening here, I suggest to rename `parentConfig` to `extendingConfig`